packages/server/src/resources/course.rs

```diff
@@ -208,6 +208,32 @@ pub struct CourseBank {
     pub credit: Option<f32>,
 }
 
+fn replace_occurrences(vec: &mut [CourseId], course: &CourseId, replacement: &CourseId) {
```

Comment:
If this function is only used inside `CourseBank::replace_course`, we should probably move it there (we can declare functions inside of other functions in rust).

Also, rename `vec` parameter to something more meaningful like `course_ids`